File: example-java/app/src/main/java/com/akaita/android/morphview/example/java/FavoriteToSendMorphView.java
package com.akaita.android.morphview.example.java;

/**
 * Created by mikel on 27/03/2018.
 */

import android.content.Context;
import android.support.annotation.Nullable;
import android.support.graphics.drawable.AnimatedVectorDrawableCompat;
import android.util.AttributeSet;

public class FavoriteToSendMorphView extends android.support.v7.widget.AppCompatImageView {

    @Nullable
    private AnimatedVectorDrawableCompat sendToFavorite;
    @Nullable
    private AnimatedVectorDrawableCompat favoriteToSend;
    private boolean showingFavorite = false;

    public FavoriteToSendMorphView(Context context) {
        super(context);
        init();
    }

    public FavoriteToSendMorphView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public FavoriteToSendMorphView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        showingFavorite = true;
        favoriteToSend = AnimatedVectorDrawableCompat.create(getContext(), R.drawable.avd_favorite_to_send);
        sendToFavorite = AnimatedVectorDrawableCompat.create(getContext(), R.drawable.avd_send_to_favorite);
        setImageDrawable(favoriteToSend);
    }

    public void showFavorite() {
        if (!showingFavorite) {
            morph();
        }
    }

    public void showSend() {
        if (showingFavorite) {
            morph();
        }
    }

    public void morph() {
        AnimatedVectorDrawableCompat drawable = showingFavorite ? favoriteToSend : sendToFavorite;
        setImageDrawable(drawable);
        if (drawable != null) {
            drawable.start();
        }
        showingFavorite = !showingFavorite;
    }

}

File: example-kotlin/app/src/main/java/com/akaita/android/morphview/example/kotlin/FavoriteToSendMorphView.kt
package com.akaita.android.morphview.example.kotlin

/**
 * Created by mikel on 26/03/2018.
 */

import android.content.Context
import android.support.graphics.drawable.AnimatedVectorDrawableCompat
import android.util.AttributeSet

class FavoriteToSendMorphView @JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : android.support.v7.widget.AppCompatImageView(context, attrs, defStyleAttr) {

    private var sendToFavorite: AnimatedVectorDrawableCompat? = null
    private var favoriteToSend: AnimatedVectorDrawableCompat? = null
    private var showingFavorite: Boolean = false

    init {
        showingFavorite = true
        favoriteToSend = AnimatedVectorDrawableCompat.create(context, R.drawable.avd_favorite_to_send)
        sendToFavorite = AnimatedVectorDrawableCompat.create(context, R.drawable.avd_send_to_favorite)
        setImageDrawable(favoriteToSend)
    }

    fun showFavorite() {
        if (!showingFavorite) {
            morph()
        }
    }

    fun showSend() {
        if (showingFavorite) {
            morph()
        }
    }

    fun morph() {
        val drawable = if (showingFavorite) favoriteToSend else sendToFavorite
        setImageDrawable(drawable)
        drawable?.start()
        showingFavorite = !showingFavorite
    }
}
